tests/plugins/dialog/focus.js

```diff
@@ -290,6 +291,197 @@
 				.then( assertFocusedTab( 'hp-test2' ) )
 				.then( focusElement( { key: KEYS.ARROW_RIGHT } ) )
 				.then( assertFocusedTab( 'hp-test4' ) );
+		},
+
+		'test moving the focus from input field to the first element of the radio group by pressing TAB key when no elements from the radio group was selected': function() {
+			var bot = this.editorBot;
+
+			return bot.asyncDialog( 'singleRadioGroupDialog' )
+				.then( assertFocusedElement( {
+					tab: 'srg-test1',
+					elementId: 'srg-input1'
+				} ) )
+				.then( focusElement( { key: KEYS.TAB } ) )
+				.then( assertFocusedElement( {
+					tab: 'srg-test1',
+					radioGroupId: 'srg-radio1',
+					radioElementIndex: 0
+				} ) );
+		},
+
+		'test moving the focus to the last element of the radio group by pressing SHIFT + TAB key': function() {
+			var bot = this.editorBot;
+
+			return bot.asyncDialog( 'singleRadioGroupDialog' )
+				.then( assertFocusedElement( {
+					tab: 'srg-test1',
+					elementId: 'srg-input1'
+				} ) )
+				.then( focusElement( { key: KEYS.TAB, shiftKey: true } ) )
+				.then( focusElement( { key: KEYS.TAB, shiftKey: true } ) )
+				.then( focusElement( { key: KEYS.TAB, shiftKey: true } ) )
+				.then( focusElement( { key: KEYS.TAB, shiftKey: true } ) )
+				.then( assertFocusedElement( {
+					tab: 'srg-test1',
+					radioGroupId: 'srg-radio1',
+					radioElementIndex: 3
+				} ) );
+		},
+
+		'test moving the focus from input field to previously checked radio group element by pressing TAB key': function() {
+			var bot = this.editorBot;
+
+			return bot.asyncDialog( 'singleRadioGroupDialog' )
+				.then( assertFocusedElement( {
+					tab: 'srg-test1',
+					elementId: 'srg-input1'
+				} ) )
+				.then( checkRadioGroupElement( {
+					tab: 'srg-test1',
+					radioGroupId: 'srg-radio1',
+					radioElementIndex: 2
+				} ) )
+				.then( focusElement( { key: KEYS.TAB } ) )
+				.then( assertFocusedElement( {
+					tab: 'srg-test1',
+					radioGroupId: 'srg-radio1',
+					radioElementIndex: 2
+				} ) );
+		},
+
+		'test moving the focus from the last radio group item to the next dialog element by pressing TAB key': function() {
+			var bot = this.editorBot;
+
+			if ( CKEDITOR.env.ie ) {
+				assert.ignore();
+			}
```

Comment:
There was probably an issue with checking radio group elements via code. Looks, that focus is not updating by using our API in unit tests. I can't confirm this because I've no way to open the console on the IE browser.